assets/src/edit-story/components/panels/textStyle/font.js

```diff
@@ -83,12 +85,56 @@ function FontControls({ selectedElements, pushUpdate }) {
       fonts,
     })
   );
+
   const fontWeights = useMemo(() => getFontWeights(getFontByName(fontFamily)), [
     getFontByName,
     fontFamily,
   ]);
+
   const fontStyle = isItalic ? 'italic' : 'normal';
 
+  useEffect(() => {
+    // only check if the current fontWeight is supported when it has changed to avoid unnecessary rechecks
+    if (fontWeight !== fontWeightRef?.current) {
```

Comment:
Hm, I'm still unsure how this is supposed to handle multiple font weights?

And it doesn't seem to:

1. Add a new text field with default Roboto font
1. Make some of it "light" font-weight (only some, not all the text)
1. Change the font family to Alegreya
1. Notice that the entire font now appears uniform (because Alegreya doesn't have "light"), but font-weight dropdown still says "(multiple)".
1. Switch back to Roboto font family
1. Notice that the light part of the text is still light, meaning that no conversion was made when switching to Alegreya.

Reply:
i had intentionally thought it would be nice to maintain the weight sets of the text element, but the more i think about what you are saying - you're of the mindset that when we swap a font family the font weight should always reset to 400/regular (or if that's not available then the closest weight to that). 

Given Omar's comment that I linked below - if we can all get on the same page for how we want to reset then I can take a 3rd try at this (3rd time's the charm, right ;) ) - I'm just confused I think by my lack of context with whatever offline conversation/history has happened within this functionality, so thanks for your patience and help.